packages/contracts/deploy/new/10_framework/01_ens_subdomains.ts

```diff
@@ -1,13 +1,45 @@
-import {DeployFunction} from 'hardhat-deploy/types';
-import {HardhatRuntimeEnvironment} from 'hardhat/types';
+import {ENSRegistry, ENSRegistry__factory} from '../../../typechain';
 import {
+  ENS_ADDRESSES,
   getContractAddress,
   getENSAddress,
   getPublicResolverAddress,
   registerSubnodeRecord,
   transferSubnodeChain,
 } from '../../helpers';
-import {ENSRegistry__factory} from '../../../typechain';
+import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers';
+import {DeployFunction} from 'hardhat-deploy/types';
+import {HardhatRuntimeEnvironment} from 'hardhat/types';
+
+async function registerAndTransferDomain(
+  ensRegistryContract: ENSRegistry,
+  managingDAOAddress: string,
+  domain: string,
+  node: string,
+  deployer: SignerWithAddress,
+  hre: HardhatRuntimeEnvironment,
+  ethers: any
+) {
+  let owner = await ensRegistryContract.owner(node);
```

Comment:
What happens if `owner = address(0)`? Previously, we called `registerSubnodeRecord`. What happens now?

Reply:
The previous way caused lots of edge case problems which I don't fully remember now.

Currently, if `owner` is `address(0)` it will fail. We explain this in readme that all domains must be owned by `deployer`. I think that's much better than to have edge case problems.